### PR TITLE
feat(CORE):Add getter for completed achievement map in AchievementManager

### PR DESCRIPTION
**Pull request by Tralenor**
## Changes Proposed:
-  Add a Function on Achievement Manager to get read-only access to the map which contains all completed achievements. This allows for easier access to the Information of a players completed achievements. 

## Tests Performed:

- Does it build without errors? yes!
- Did you test in-game?  yes, with a custom module where I listened to the OnAchiCompleted Hook an then accessed the map over:  auto completedAchievements = player->GetAchievementMgr()->GetCompletedAchievements();
- On which OS did you test?  Ubuntu and Debian
- Describe any other tests performed --> none.

## How to Test the Changes:

No functionality change out of the box, just a "Getter"-Fucntion to allow modules easier access to Information about a Players achieved Achievements - thus there is nothing no test out of the box. 
An example on how this could be used:     
```c++
void OnAchiComplete(Player *player, AchievementEntry const *achievement) override {
        auto completedAchievements = player->GetAchievementMgr()->GetCompletedAchievements();

        auto achievementsPoints = std::accumulate(completedAchievements.cbegin(), completedAchievements.cend(), 0,
                                                  [](int sum, auto element) {
                                                      return sum + sAchievementStore.LookupEntry(element.first)->points;
                                                  });

        std::stringstream message{};
        message << "Achievement gained, your new total is: " << achievementsPoints;

        player->GetSession()->SendAreaTriggerMessage("%s", message.str().c_str());
    }
```

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 1]
```diff
@@ -2394,6 +2394,10 @@ bool AchievementMgr::CanUpdateCriteria(AchievementCriteriaEntry const* criteria,
     return true;
 }
 
+const CompletedAchievementMap& AchievementMgr::GetCompletedAchievements() {
```
Comment (Kitzunu) [inline src/server/game/Achievements/AchievementMgr.cpp:2397]:
```suggestion
CompletedAchievementMap const& AchievementMgr::GetCompletedAchievements()
{
```

You can read more about the C++ standards we use here: https://www.azerothcore.org/wiki/cpp-code-standards

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/game/Achievements/AchievementMgr.h [hunk 2]
```diff
@@ -300,6 +300,7 @@ class AchievementMgr
 
     void RemoveCriteriaProgress(AchievementCriteriaEntry const* entry);
     CriteriaProgress* GetCriteriaProgress(AchievementCriteriaEntry const* entry);
+    const CompletedAchievementMap& GetCompletedAchievements();
```
Comment (Kitzunu) [inline src/server/game/Achievements/AchievementMgr.h:303]: and here :D

Comment (Tralenor) [reply src/server/game/Achievements/AchievementMgr.h:303]: ouch - gonna fix it, 1sec

File: src/server/game/Achievements/AchievementMgr.cpp (hunk 1 above)
Comment (Tralenor) [reply src/server/game/Achievements/AchievementMgr.cpp]: thank you, is there somekind of linter I could use to enforce them? (or at least parts of it?)

File: src/server/game/Achievements/AchievementMgr.h (hunk 2 above)
Comment (Tralenor) [reply src/server/game/Achievements/AchievementMgr.h]: should be fixed now.

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**